cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -232,18 +233,8 @@ public BodyConsumer createAppVersion(HttpRequest request, HttpResponder responde
                                        @PathParam("version-id") final String versionId)
     throws Exception {
 
-    ApplicationId applicationId = validateApplicationVersionId(namespaceId, appId, versionId);
-
-    if (!applicationLifecycleService.updateAppAllowed(applicationId)) {
-      responder.sendString(HttpResponseStatus.CONFLICT,
-                           String.format("Cannot update the application because version %s already exists", versionId));
-    }
-    try {
-      return deployAppFromArtifact(applicationId);
-    } catch (Exception ex) {
-      responder.sendString(HttpResponseStatus.INTERNAL_SERVER_ERROR, "Deploy failed: " + ex.getMessage());
-      return null;
-    }
+    // Ignore the version provided by the user. Treating it the same as deploy without version
+    return create(request, responder, namespaceId, appId);
```

Comment:
I believe this method should be renamed as it does not do create, it allows to deploy new version irrespective if this is new or existing app, correct?

Reply:
Also please change the method to take namespace + application name instead of app id as it does not use the version

Reply:
renamed this function to `deploy`  
The variable name appId here is kinda misleading, it is actually a string for application name, not an `ApplicationId`